### PR TITLE
logs stdout, stderr and integration tests

### PR DESCRIPTION
**Pull request by amihaiemil**
PR for #256. Apparently, it is mandatory to specify either ``stdout`` or ``stderr`` when reading the logs. Left some puzzles.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#257](https://github.com/amihaiemil/docker-java-api/issues/257) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#257](https://github.com/amihaiemil/docker-java-api/issues/257) is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/G6LPQQV2P/84a03d82-0e75-4de2-bd7c-2b3545a431fe); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @amihaiemil[/z](https://www.0crat.com/u/amihaiemil) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/test/java/com/amihaiemil/docker/RtLogsITCase.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import org.hamcrest.Matchers;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import java.io.File;
+
+/**
+ * Integration tests for {@link RtLogs}.
+ * @author Mihai Andronache (amihaiemil@gmail.com)
+ * @version $Id#
```
Comment (paulodamaso) [inline src/test/java/com/amihaiemil/docker/RtLogsITCase.java]: @amihaiemil `$Id$` here

File: src/test/java/com/amihaiemil/docker/RtLogsITCase.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+     */
+    @Test
+    public void fetchesLogs() throws Exception {
+        final Container container =  new LocalDocker(
+            new File("/var/run/docker.sock")
+        ).images().pull("hello-world", "latest").run();
+        final String logs = container.logs().fetch();
+        MatcherAssert.assertThat(
+            logs,
+            Matchers.not(Matchers.isEmptyOrNullString())
```
Comment (paulodamaso) [inline src/test/java/com/amihaiemil/docker/RtLogsITCase.java]: @amihaiemil I'd prefer to use non-static matchers, makes the code more elegant

File: src/test/java/com/amihaiemil/docker/RtLogsITCase.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        );
+        MatcherAssert.assertThat(
+            logs.trim(),
+            Matchers.startsWith("Hello from Docker!")
```
Comment (paulodamaso) [inline src/test/java/com/amihaiemil/docker/RtLogsITCase.java]: @amihaiemil Same as above

File: src/test/java/com/amihaiemil/docker/RtLogsITCase.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    @Test
+    @Ignore
+    public void followsLogs() throws Exception {
+        final Container container =  new LocalDocker(
+            new File("/var/run/docker.sock")
+        ).images().pull("hello-world", "latest").run();
+        final String logs = IOUtils.toString(container.logs().follow());
+        MatcherAssert.assertThat(
+            logs,
+            Matchers.not(Matchers.isEmptyOrNullString())
```
Comment (paulodamaso) [inline src/test/java/com/amihaiemil/docker/RtLogsITCase.java]: @amihaiemil Here too, no static matchers

File: src/test/java/com/amihaiemil/docker/RtLogsITCase.java [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        );
+        MatcherAssert.assertThat(
+            logs.trim(),
+            Matchers.startsWith("Hello from Docker!")
```
Comment (paulodamaso) [inline src/test/java/com/amihaiemil/docker/RtLogsITCase.java]: @amihaiemil Here again

**Comment by paulodamaso** [comment]
@rultor merge

**Comment by amihaiemil** [comment]
@rultor merge it

**Comment by 0crat** [comment]
Job was finished in 0 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
The job [#257](https://github.com/amihaiemil/docker-java-api/issues/257) is now out of scope

**Comment by 0crat** [comment]
Order was finished: +20 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)

**Pull request merged**